### DISCUSSION TITLE
reload content scripts - help

### DISCUSSION THREAD

**Discussion by KULTI1995** [wxt-dev/wxt#1003] [Q&A]
"Reloaded: content" - How to disable page reload when we change contentscripts? I want to implement it myself, manage it.

**Comment by aklinker1** [comment]
There isn't a way right now. What's your use case? Maybe I can add a option to disable it

**Comment by KULTI1995** [comment]
@aklinker1 Any hack to do this?

Well, I once implemented hot reload on another "boilerplate" without reloading the page. I want to implement hot reload, so that it reloads the script, replaces the script after the update with the new version on the open site. Styling and programming, this is a more convenient solution. Then it would remove the react from the container the previous script and place the updated one, without reloading the page. To keep the application states the same, you can implement something like this:
https://www.joshwcomeau.com/react/persisting-react-state-in-localstorage/

Then on the production version, it will be a regular setState, and on the dev version, it will be saving the state, on the current page. So that when the script is refreshed, the application state remains. This is very easy to implement. I only need access to this, to not allow the refresh, or instead of refreshing, to send information about the update to the background, instead of refreshing the page.

**Reply by aklinker1** [reply]
Patch the package. Here's the code to replace:

https://github.com/wxt-dev/wxt/blob/e3f00d279781c03dfd0903b41aada100ae5b1d55/packages/wxt/src/virtual/background-entrypoint.ts#L15-L17

Should be located in `node_modules/wxt/dist/virtual/background-entrypoint.js`

The dev server sends this event to the background. You can follow the `reloadContentScript` definition to see the call to reload the tabs.

**Comment by KULTI1995** [comment]
When can we expect an update? I'm currently creating an interesting browser add-on based on your boilerplate. And live refreshing makes coding much faster.

**Reply by aklinker1** [reply]
There's already an issue around [ESM content scripts and HMR](https://github.com/wxt-dev/wxt/issues/357)... but there are problems. So a built-in solution for what you're describing won't be coming any time soon.

I also hesitent to about adding a way to customize the runtime reload behavior... Would need to come up with a future proof solution that doesn't get replaced when that issue is implemented. So no estimate
